### DISCUSSION TITLE
How to I Change RunSpeed in Runtime?? (Use c++)

### DISCUSSION THREAD

**Discussion by tlsduf** [Sixze/ALS-Refactored#557] [Q&A] [answered]
I would like to change the character's movement speed during runtime in C++ code. I found out that FAlsMovementGaitSettings initializes the character's movement speed, but I couldn't find a function called setSpeed in the code. Please help!

**Comment by tlsduf** [comment]
I just resolved the issue! I copied MS_Als_Normal a few times, renamed them as FT_MS_Als_Normal_slope1,2,3, etc., and modified the speed in the respective data. Then, I added a variable to the character and assigned it. Now, using the AlsCharacterMovement->SetMovementSettings(MovementSettingsSlope1) function, I can assign the MovementSettings according to the conditions. Good luck, everyone!

**Reply by ameaninglessname** [reply, marked as answer]
Glad to see that, you could mark this as resolved.
